### PR TITLE
Implement Google Places API

### PR DESCRIPTION
**Pull request by pimolo**
## TODO
- [x] Implement Details API
- [x] Implement Search API
- [x] Controller
## Description

To call this API i decided to use [this lib](https://github.com/Nekland/PlacesApi).
I created data transformers to turn the data fetched from Google into a PHP object which is simpler to manipulate.
So when you call the Details API, you actually get a PHP object 😸

### CODE REVIEW

File: src/UnderDevCrew/Bundle/Transformer/PlacesModelTransformer.php [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+    /**
+     * @param PlacesModelFactory $placesModelFactory
+     */
+    public function __construct(PlacesModelFactory $placesModelFactory)
+    {
+        $this->modelFactory = $placesModelFactory;
+    }
+
+
+    public function transform($data, $type = self::UNKNOWN)
```
Comment (pimolo) [inline src/UnderDevCrew/Bundle/Transformer/PlacesModelTransformer.php]: {@inheritDoc}

File: src/UnderDevCrew/Bundle/Transformer/PlacesModelTransformer.php [hunk 2]
```diff
@@ -0,0 +1,29 @@
...
+
+    /**
+     * @param PlacesModelFactory $placesModelFactory
+     */
+    public function __construct(PlacesModelFactory $placesModelFactory)
+    {
+        $this->modelFactory = $placesModelFactory;
+    }
+
+
```
Comment (pimolo) [inline src/UnderDevCrew/Bundle/Transformer/PlacesModelTransformer.php]: missing {@inheritDoc}

File: src/UnderDevCrew/Bundle/Model/PlacesModel.php [hunk 3]
```diff
@@ -0,0 +1,223 @@
...
+
+class PlacesModel
+{
+    /**
+     * @var string
+     */
+    private $id;
+
+    /**
+     * @var string
```
Comment (girard-g) [inline src/UnderDevCrew/Bundle/Model/PlacesModel.php]: Why not a float ? Same for longitude

Comment (pimolo) [inline src/UnderDevCrew/Bundle/Model/PlacesModel.php]: Why not :) Moreover it's considered as a float by Google Places itself (according to the format delivered in the response). Nice catch!

File: src/UnderDevCrew/Bundle/Transformer/PlacesModelTransformer.php [hunk 4]
```diff
@@ -0,0 +1,29 @@
...
+    public function transform($data, $type = self::UNKNOWN)
+    {
+        $jsonData = json_decode($data, true);
```
Comment (girard-g) [inline src/UnderDevCrew/Bundle/Transformer/PlacesModelTransformer.php]: For some readable reasons, can you change your var name into something like $decodedJsonData ? I think this is something witch can render a hard understanding code into a stuff more user friendly. As if I read the code I can stop at the var name to understand what it contend. Of course this is detail.

**Comment by girard-g** [comment]
:+1:

**Comment by girard-g** [comment]
Seems legit ! good job !

**Comment by duncanroze** [comment]
👍

**Pull request merged**